doc/bn.tex

```diff
@@ -2185,12 +2185,41 @@ \section{Binary Conversions}
 byte depending on the sign.  If the sign is zpos (e.g. not negative) the prefix is zero, otherwise the prefix
 is non--zero.
 
-The two functions \texttt{mp\_import} and \texttt{mp\_export} implement the corresponding GMP functions as described at \url{http://gmplib.org/manual/Integer-Import-and-Export.html}.
-\index{mp\_import} \index{mp\_export}
+The two functions \texttt{mp\_unpack} (get your gifts out of the box, import binary data) and \texttt{mp\_pack} (put your gifts into the box, export binary data) implement the similarly working GMP functions as described at \url{http://gmplib.org/manual/Integer-Import-and-Export.html} with the exception that \texttt{mp\_pack} will not allocate memory if \texttt{rop} is \texttt{NULL}.
+\index{mp\_unpack} \index{mp\_pack}
 \begin{alltt}
-int mp_import(mp_int *rop, size_t count, int order, size_t size, int endian, size_t nails, const void *op);
-int mp_export(void *rop, size_t *countp, int order, size_t size, int endian, size_t nails, const mp_int *op);
+int mp_unpack(mp_int *rop, size_t count, mp_order order, size_t size,
+             mp_endian endian, size_t nails, const void *op, size_t maxsize, size_t *written);
+int mp_pack(void *rop, size_t *countp, mp_order order, size_t size,
+             mp_endian endian, size_t nails, const mp_int *op);
 \end{alltt}
+The function \texttt{mp\_pack} has the additional variable \texttt{maxsize} which must hold the size of the buffer \texttt{rop} in bytes.
+
+To enhance the readability of your code, the following enums have been wrought for your convenience.
+\begin{alltt}
+#ifdef MP_USE_ENUMS
+typedef enum {
+   MP_LSB_FIRST = -1,
+   MP_MSB_FIRST =  1
+} mp_order;
+typedef enum {
+   MP_LTTLE_ENDIAN  = -1,
+   MP_NATIVE_ENDIAN =  0,
+   MP_BIG_ENDIAN    =  1
+} mp_endian;
+\end{alltt}
+Also available as preprocessor macros.
+\begin{alltt}
```

Comment:
This is not 100% correct. I would not describe them as enums/ints. After 1.2 we will make MP_USE_ENUMS the default.

Reply:
Maybe also a bit too verbose docs replicating the code of a triviality.

Reply:
> Maybe also a bit too verbose docs replicating the code for a triviality.

I can squeeze it into a sentence but I find code snippets are quicker to read for a programmer. But the snippet after "Also available as preprocessor macros." can go, that's right, the sentence introducing them is sufficient.